doc/Getting_Started.md

```diff
@@ -30,8 +30,10 @@ developing Tock.
 
 1. [Rust](http://www.rust-lang.org/)
 2. [rustup](https://rustup.rs/) to install Rust (version >= 1.11.0)
-3. Command line utilities: make
-4. A supported board or QEMU configuration.
+3. Host toolchain (gcc, glibc)
```

Comment:
I might be wrong, but I don't think the actual host toolchain is needed. I think the only requirements are the build utilities (find, make, ...).

Reply:
I also have a Fedora dockerfile that I used, and it kept failing the same way until I added host gcc. Feel free to try:

```
# container for building Tock kernel
# podman build -f tock.dockerfile -t tock_kernel
# podman run -ti --name=tock_kernel -v.:/mnt/build/:z tock_kernel /bin/bash
FROM fedora:35

RUN dnf install -y arm-none-eabi-gcc-cs make gcc findutils
RUN curl https://sh.rustup.rs -sSf | sh -s -- -y
```

Sadly, the containers are big and I ran out of space, so can't double-check right now :S

Reply:
Got it. Do you still have the fail log? It would be interesting to see why.

Reply:
Those are two failure modes that I've seen, both one after another, in the Debian-bsed container.

[tock.log](https://github.com/tock/tock/files/8026285/tock.log)

Reply:
Thank you, it seems to be a linker problem. Maybe others can have a better response to this.

Reply:
The discussion at https://github.com/rust-lang/rust/issues/71515 might be informative.

I *think* that using `-Zgcc-ld=lld` might fix this so that we use the Rust-shipped lld instead of gcc's, but that is an unstable option. Unless we switch to using that, it seems accurate that we require a gcc installation, which is unfortunate.

Reply:
Am I understanding this correctly that what this means is there are some gcc files that llvm-ld is using?

Reply:
I tried changing the makefile to:

```
RUSTC_FLAGS ?= \
  -C link-arg=-Tlayout.ld \
  -C linker=rust-lld \
  -C linker-flavor=ld.lld \
  -C relocation-model=static \
  -C link-arg=-nmagic \
  -C link-arg=-icf=all \
  -Z gcc-ld=lld \
```

and got:

```
     Running `rustc --crate-name hail --edition=2021 boards/hail/src/main.rs --error-format=json --json=diagnostic-rendered-ansi --crate-type bin --emit=dep-info,link -C opt-level=z -C panic=abort -C lto -C codegen-units=1 -C debuginfo=2 -C link-arg=-L/Users/bradjc/git/tock/boards/hail -C metadata=5e232a3c154cf8d9 -C extra-filename=-5e232a3c154cf8d9 --out-dir /Users/bradjc/git/tock/target/thumbv7em-none-eabi/release/deps --target thumbv7em-none-eabi -L dependency=/Users/bradjc/git/tock/target/thumbv7em-none-eabi/release/deps -L dependency=/Users/bradjc/git/tock/target/release/deps --extern capsules=/Users/bradjc/git/tock/target/thumbv7em-none-eabi/release/deps/libcapsules-1202625586e27f87.rlib --extern 'noprelude:compiler_builtins=/Users/bradjc/git/tock/target/thumbv7em-none-eabi/release/deps/libcompiler_builtins-46d9de604032590c.rlib' --extern components=/Users/bradjc/git/tock/target/thumbv7em-none-eabi/release/deps/libcomponents-9fd8dad9e0b9bcb3.rlib --extern 'noprelude:core=/Users/bradjc/git/tock/target/thumbv7em-none-eabi/release/deps/libcore-fee7ca5f2aa3d4f2.rlib' --extern cortexm4=/Users/bradjc/git/tock/target/thumbv7em-none-eabi/release/deps/libcortexm4-b147e95b800d37f5.rlib --extern kernel=/Users/bradjc/git/tock/target/thumbv7em-none-eabi/release/deps/libkernel-dc50ffa5fc0275b6.rlib --extern sam4l=/Users/bradjc/git/tock/target/thumbv7em-none-eabi/release/deps/libsam4l-fb66e817481084b7.rlib -Z unstable-options -C link-arg=-Tlayout.ld -C linker=rust-lld -C linker-flavor=ld.lld -C relocation-model=static -C link-arg=-nmagic -C link-arg=-icf=all -Z gcc-ld=lld --remap-path-prefix=/Users/bradjc/git/tock/= --remap-path-prefix=/Users/bradjc/.rustup/toolchains/nightly-2021-12-04-x86_64-apple-darwin/lib/rustlib/src/rust/library/core=/core/`
error: option `-Z gcc-ld` is used even though linker flavor is not gcc

error: could not compile `hail` due to previous error

Caused by:
  process didn't exit successfully: `rustc --crate-name hail --edition=2021 boards/hail/src/main.rs --error-format=json --json=diagnostic-rendered-ansi --crate-type bin --emit=dep-info,link -C opt-level=z -C panic=abort -C lto -C codegen-units=1 -C debuginfo=2 -C link-arg=-L/Users/bradjc/git/tock/boards/hail -C metadata=5e232a3c154cf8d9 -C extra-filename=-5e232a3c154cf8d9 --out-dir /Users/bradjc/git/tock/target/thumbv7em-none-eabi/release/deps --target thumbv7em-none-eabi -L dependency=/Users/bradjc/git/tock/target/thumbv7em-none-eabi/release/deps -L dependency=/Users/bradjc/git/tock/target/release/deps --extern capsules=/Users/bradjc/git/tock/target/thumbv7em-none-eabi/release/deps/libcapsules-1202625586e27f87.rlib --extern 'noprelude:compiler_builtins=/Users/bradjc/git/tock/target/thumbv7em-none-eabi/release/deps/libcompiler_builtins-46d9de604032590c.rlib' --extern components=/Users/bradjc/git/tock/target/thumbv7em-none-eabi/release/deps/libcomponents-9fd8dad9e0b9bcb3.rlib --extern 'noprelude:core=/Users/bradjc/git/tock/target/thumbv7em-none-eabi/release/deps/libcore-fee7ca5f2aa3d4f2.rlib' --extern cortexm4=/Users/bradjc/git/tock/target/thumbv7em-none-eabi/release/deps/libcortexm4-b147e95b800d37f5.rlib --extern kernel=/Users/bradjc/git/tock/target/thumbv7em-none-eabi/release/deps/libkernel-dc50ffa5fc0275b6.rlib --extern sam4l=/Users/bradjc/git/tock/target/thumbv7em-none-eabi/release/deps/libsam4l-fb66e817481084b7.rlib -Z unstable-options -C link-arg=-Tlayout.ld -C linker=rust-lld -C linker-flavor=ld.lld -C relocation-model=static -C link-arg=-nmagic -C link-arg=-icf=all -Z gcc-ld=lld --remap-path-prefix=/Users/bradjc/git/tock/= --remap-path-prefix=/Users/bradjc/.rustup/toolchains/nightly-2021-12-04-x86_64-apple-darwin/lib/rustlib/src/rust/library/core=/core/` (exit status: 1)
make: *** [/Users/bradjc/git/tock/target/thumbv7em-none-eabi/release/hail] Error 101
```

I'm not sure this comment is correct (https://github.com/rust-lang/rust/issues/39915#issuecomment-998789288) because we use `-C linker=rust-lld` on mac and linux.

Anyway I think the issue is overriding the `VERBOSE` variable used inside the makefile. When I do that I can't get the kernel to build and get similar errors.